### PR TITLE
Remove CLI ESLint suppressions (Fixes #2086)

### LINKED ISSUE #2086: Remove inline ESLint disables and warnings: packages/cli commands/services/config
Parent: ESLint complexity-suppression cleanup and enforcement.

## Goal

Remove the file-level complexity/size suppressions from **packages/cli** command, service, and config modules by decomposing them.

## Files in scope

    packages/cli/src/config/profileRuntimeApplication.ts
    packages/cli/src/services/McpPromptLoader.ts
    packages/cli/src/ui/commands/diagnosticsCommand.ts
    packages/cli/src/ui/commands/mcpCommand.ts
    packages/cli/src/ui/commands/memoryCommand.ts
    packages/cli/src/ui/commands/profileCommand.ts
    packages/cli/src/ui/commands/schema/index.ts
    packages/cli/src/ui/commands/setCommand.ts
    packages/cli/src/ui/commands/statsCommand.ts
    packages/cli/src/ui/commands/todoCommand.ts

## How to approach

For each file: remove the file-level `/* eslint-disable complexity, sonarjs/cognitive-complexity, max-lines */` directive and make the file pass by **decomposition**, not by re-suppressing:
- Extract cohesive helpers / sub-modules (pure functions, smaller classes) so each unit is under the budget (complexity 25, cognitive-complexity 30, max-lines 800, max-lines-per-function 80).
- Keep public APIs and exported symbols stable. Imports elsewhere must not change behavior.
- Preserve existing tests; they must pass unchanged. Add focused tests only if you extract a unit that warrants direct coverage.

## Hard rules

- **No behavioral changes.** Structure-only refactor.
- **Do not relocate the disable to a narrower scope** to "pass". The file must actually meet the budget. A genuinely unavoidable single line may keep a line-level disable, but only with a specific, real rationale (never "Phase 5").
- No new file-level complexity/`max-lines` disables.
- Verify before PR: `npm run test`, `npm run lint`, `npm run typecheck`, `npm run format`, `npm run build`.

## Acceptance criteria

- Every file listed below no longer has a file-level complexity/cognitive-complexity/max-lines disable.
- `npm run lint` passes for these files with the budget enforced.
- No behavioral regressions; existing tests green.

---

## Policy update for this cleanup scope

This sub-issue is part of #2078's stricter end state. It is **not** limited to file-level complexity suppressions anymore.

Within this issue's scope, the implementing agent must remove all inline ESLint disable directives and fix all lint warnings/errors that appear when the scope is un-excluded.

Required behavior:

1. At the start of the work, remove or narrow any temporary `eslint.config.js` exclusion that covers this scope.
2. Run lint for this scope and expose the actual failures.
3. Fix the code/tests instead of adding suppressions.
4. Leave the scope locked so new inline disables fail.

Do not add a baseline allow-list. Do not keep Phase 5 suppressions. Do not add new inline `eslint-disable`, `eslint-disable-next-line`, or `eslint-disable-line` comments.

Fail-fast guidance:

- For `@typescript-eslint/no-unnecessary-condition`, do not preserve defensive-spackle checks that contradict the static type. If external data is uncertain, validate or parse it at the boundary and throw early; then make downstream types accurate.
- For `@typescript-eslint/prefer-nullish-coalescing`, if empty string/zero/false must fall through, write that explicitly with a named helper or direct condition. Do not suppress the rule.
- For `@typescript-eslint/no-explicit-any`, type the value. In tests, use shared helpers or narrowly scoped config only if there is a real mocking need.
- For regex warnings, fix the regex or use a narrowly scoped config-level exclusion only for genuinely non-applicable cases such as terminal/ANSI control-character parsing.

Project-wide decision inherited from #2079:

- `no-os-command-from-path` / `sonarjs/no-os-command-from-path` and equivalent SonarJS os-command PATH/tool invocation warnings should be disabled centrally for the project, not suppressed inline in this scope.

Acceptance criteria for this scope:

- No inline ESLint disable directives remain in files covered by this sub-issue.
- No lint warnings remain for files covered by this sub-issue.
- Any permanent exception is in `eslint.config.js`, tightly scoped, and justified.
- Full project verification passes before the PR is considered complete.

### PR DESCRIPTION
**Pull request by acoliver**
## TLDR

Removes the CLI complexity/max-lines ESLint suppressions for issue #2086 by decomposing large command/service/config modules into focused helper modules, adding behavior-focused tests for extracted parsing and guard logic, and locking the completed scope against future inline disable directives.

## Dive Deeper

This PR refactors the ten #2086 CLI files without intended behavior changes:

- Extracts MCP prompt argument parsing, diagnostics token rendering, MCP display/auth handling, profile load/load-balancer/schema helpers, command schema helpers, set-command schema construction, stats quota formatting, and todo operation/formatting helpers.
- Removes file-level and inline ESLint disable directives from the #2086 target files and extracted helpers.
- Adds completed-scope locking in eslint.config.js for the target files and helper modules.
- Keeps the broad CLI legacy directive scope available for other cleanup issues while overriding it for #2086-complete files.
- Adds tightly scoped config-level SonarJS regex exceptions for the extracted argument parsers only, with #2086 policy comments.
- Adds focused tests for parser extraction and behavior-preservation edge cases.

## Reviewer Test Plan

Recommended validation:

1. Inspect eslint.config.js and confirm #2086 target/helper files are listed under completed directive cleanup scopes.
2. Confirm no inline ESLint disable directives remain in the #2086 target/helper files.
3. Run npm run lint and verify complexity/cognitive/max-lines budgets are enforced for the refactored files.
4. Exercise profile save parsing edge cases:
   - model quoted names with spaces
   - quoted names with bucket args
   - malformed adjacent quote text preserving legacy fallback
5. Exercise stats command behavior with incomplete runtime stats to confirm it returns the existing error item rather than throwing.

## Testing Matrix

|          | 🍏  | 🪟  | 🐧  |
| -------- | --- | --- | --- |
| npm run  | ✅  | ❓  | ❓  |
| npx      | ✅  | ❓  | ❓  |
| Docker   | ❓  | ❓  | ❓  |
| Podman   | ❓  | -   | -   |
| Seatbelt | ❓  | -   | -   |

Validated on macOS:

- npm run test
- npm run lint
- npm run typecheck
- npm run format
- npm run build
- node scripts/start.js --profile-load ollamaglm51 "write me a haiku and nothing else"

Note: node scripts/start.js --profile-load synthetic "write me a haiku and nothing else" was also attempted and failed with an OpenAI API communication error before producing a response; the ollamaglm51 project smoke profile passed.

## Linked issues / bugs

Fixes #2086

### CODE REVIEW

File: packages/cli/src/services/mcpPromptArgParser.test.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  it('ignores empty-string named values', () => {
+    const argValues = new Map([['a', '']]);
+    const result = assignArgsToPrompt(
+      argValues,
+      ['filled'],
+      [{ name: 'a', required: true }],
+    );
+    expect(result).toStrictEqual({ a: 'filled' });
+  });
+});
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/services/mcpPromptArgParser.test.ts:136]: Addressed in 4af49a7 by adding a regression test for the single required argument path with no named or positional input. The test asserts that the parser returns the missing required argument error instead of accepting an empty value.

File: packages/cli/src/services/mcpPromptArgParser.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+  const unfilledArgs = selectUnfilledRequiredArgs(promptArgs, promptInputs);
+  if (unfilledArgs.length === 0) {
+    return promptInputs;
+  }
+
+  if (unfilledArgs.length === 1) {
+    // Single unfilled required arg: join all positional tokens as one value.
+    promptInputs[unfilledArgs[0].name] = positionalTokens.join(' ');
+    return promptInputs;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/services/mcpPromptArgParser.ts:123]: Addressed in 4af49a7. The single required argument branch now checks for an empty positional token list and returns the missing required argument error before assigning any value.

File: packages/cli/src/ui/commands/diagnosticsTokens.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+    diagnostics.push('### Provider Tokens');
+    diagnostics.push(`- ${provider}:`);
+    diagnostics.push(`  - Buckets: ${buckets.length}`);
+
+    for (const bucket of buckets) {
+      const token = (await tokenStore.getToken(
+        provider,
+        bucket,
+      )) as ProviderToken | null;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/ui/commands/diagnosticsTokens.ts]: Addressed in 4af49a7. Provider token reads are now handled per bucket, so a failure for one provider bucket is logged and does not abort the rest of diagnostics token rendering.

File: packages/cli/src/ui/commands/profileLoad.ts [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  providerName: string,
+): Promise<void> {
+  logger.debug(
+    () =>
+      `[profile] forcing config provider manager switch to '${providerName}'`,
+  );
+  try {
+    void providerManager.setActiveProvider(providerName);
+    logActiveProviderName(providerManager);
+  } catch (error) {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/ui/commands/profileLoad.ts:95]: Addressed in 4af49a7. Provider switching is now awaited so promise rejections are caught by the surrounding error handling and the active provider log runs after the switch completes.

File: packages/cli/src/ui/commands/schema/index.ts [hunk 5]
```diff
@@ -309,32 +100,32 @@ function flattenSchemaPaths(
...
+    } else if (nextNode.kind === 'literal') {
+      // Recursively flatten nested literals
+      const nestedLiterals = nextNodes.filter(
+        (n): n is LiteralArgument => n.kind === 'literal',
+      );
+      const nestedPaths = flattenSchemaPaths(nestedLiterals, nodePath);
+      flattened.push(...nestedPaths);
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/ui/commands/schema/index.ts:127]: Addressed in 4af49a7. Nested literal flattening now performs the recursive literal pass once instead of repeating the same work for every literal in the loop.

File: packages/cli/src/ui/commands/statsCommand.nullGuard.test.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2025 Vybestack LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { vi, describe, it, expect, beforeEach } from 'vitest';
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/ui/commands/statsCommand.nullGuard.test.ts]: Addressed in 4af49a7. The null guard test now restores real timers after each test to avoid fake timer state leaking into later tests.

File: packages/cli/src/ui/commands/statsQuota.ts [hunk 7]
```diff
@@ -0,0 +1,443 @@
...
+  let provider: 'zai' | 'synthetic' | 'chutes' | 'kimi' | null = null;
+  let baseUrlForFetch: string | undefined;
+  const activeProviderName = runtimeApi.getActiveProviderName();
+
+  // Strategy 1: Check ephemeral base-url setting (highest priority)
+  const ephemeralBaseUrl = runtimeApi.getEphemeralSetting('base-url');
+  if (typeof ephemeralBaseUrl === 'string') {
+    provider = detectApiKeyProvider(ephemeralBaseUrl);
+    baseUrlForFetch = ephemeralBaseUrl;
+    if (provider) {
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/ui/commands/statsQuota.ts:146]: Addressed in 4af49a7. The ephemeral base URL is only carried forward when provider detection succeeds, so fallback provider-name detection no longer receives an unrelated URL.

File: packages/cli/src/ui/commands/todoOperations.ts [hunk 8]
```diff
@@ -0,0 +1,483 @@
...
+      return {
+        parentIndex,
+        subtaskIndex: parent.subtasks?.length ?? 0,
+        isLast: true,
+      };
+    }
+
+    // Line 59: INSERT at subtask_pos - 1
+    const subtaskIndex = parseInt(subtaskMatch[2], 10) - 1;
+    return { parentIndex, subtaskIndex, isLast: false };
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/ui/commands/todoOperations.ts:111]: Addressed in c92b668. Dotted subtask positions are now bounds-checked, including zero positions such as 1.0, and regression tests cover add and remove paths so invalid positions cannot splice from the end.

File: packages/cli/src/ui/commands/profileLoad.ts [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+  context: CommandContext,
+  result: { providerName?: string },
+): Promise<void> {
+  const configService = context.services
+    .config as unknown as ProfileConfigService | null;
+  if (configService === null) {
+    return;
+  }
+
+  const providerManager = configService.getProviderManager?.();
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/ui/commands/profileLoad.ts]: Addressed in c92b668. The config service lookup now treats null and undefined as absent before accessing getProviderManager.

File: packages/cli/src/ui/commands/statsQuota.ts [hunk 10]
```diff
@@ -0,0 +1,443 @@
...
+      : new Map<string, Record<string, unknown>>();
+  const codexUsageInfo: UsageMap =
+    codexResult.status === 'fulfilled'
+      ? codexResult.value
+      : new Map<string, Record<string, unknown>>();
+  const geminiUsageInfo: UsageMap =
+    geminiResult.status === 'fulfilled'
+      ? geminiResult.value
+      : new Map<string, Record<string, unknown>>();
+
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/ui/commands/statsQuota.ts:386]: Addressed in c92b668. Gemini quota fetch rejections are now logged with the same warning pattern used for Anthropic and Codex failures.

File: packages/cli/src/ui/commands/todoFormatters.ts [hunk 11]
```diff
@@ -0,0 +1,201 @@
...
+    const todos: Todo[] = JSON.parse(content);
+
+    const counts = countStatuses(todos);
+    const firstTitle = todos[0]?.content || '(empty)';
+    const age = formatAge(file.mtime);
+    const statusSummary = buildStatusSummary(counts);
+
+    return [
+      `${idx + 1}. ${age} │ ${todos.length} items (${statusSummary})`,
+      `   → "${firstTitle}"`,
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/ui/commands/todoFormatters.ts:107]: Addressed in c92b668. Session formatting now appends a status summary only when it is non-empty, avoiding output such as 0 items followed by empty parentheses.

File: packages/cli/src/ui/commands/todoOperations.ts [hunk 12]
```diff
@@ -0,0 +1,483 @@
...
+  const parsed = parsePosition(posStr, todos);
+  const addItem = context.ui.addItem;
+
+  if (!validateExistingParent(addItem, parsed, todos, posStr)) return;
+  if (!rejectSubtaskPosition(addItem, parsed)) return;
+
+  const newTodos = [...todos];
+  const todo = newTodos[parsed.parentIndex];
+  todo.status = newStatus;
+
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/ui/commands/todoOperations.ts:211]: Addressed in c92b668. Status updates now clone the target todo object before changing its status, so the original todo object reference from context state is not mutated.
